torch/_dynamo/variables/builder.py

```diff
@@ -2281,6 +2280,11 @@ def handle_traced_output(example_value, tx, proxy, options, subclass_type, targe
         # tensor, the stored example value will update too!)
         example_value = _clone_input(example_value, tx.fake_mode)
         set_example_value(proxy.node, example_value)
+        # We bind the unbacked symints in sizes/trdies of tensor lazily.
+        # So that subgraphs can access the unbacked symbol's proxy in parent graph
+        # when lifting unbacked symbols of input tensors to subgraph inputs.
+        # We do it lazily because the tensor may not be used in subgraphs.
+        tx.output.current_tracer.lazy_bind_unbacked_symbols(example_value, proxy)
```

Comment:
Calling this something like "track_unbacked_symbols" and changing the comment to "We have an invariant that current_tracer.bound_symbols tracks all symints in the current tracer. If we see an unbacked symbol, we need to update current_tracer.bound_symbols" is probably sufficient. The laziness is an implementation detail that we don't need to talk about here (it makes things more confusing to read)